lib/elftools/elf_file.rb

```diff
@@ -321,6 +333,54 @@ def save(filename)
       IO.binwrite(filename, all)
     end
 
+    def rebuild
+      all = header.to_binary_s
+
+      sections.each do |s|
+        s.rebuild
+
+        next if s.size <= 0
+
+        if s.header.sh_addralign != 0
+          # align with 0 bytes
+          all += "\0" * (-all.size % s.header.sh_addralign)
+        end
+
+        s.header.sh_offset = all.size
+        all += s.data
+      end
+
+      header.e_shnum = sections.size
+      header.e_shoff = all.size
+
+      sections.each do |s|
+        sh = s.header.to_binary_s
+        all += sh
+      end
+
+      all[0...header.num_bytes] = header.to_binary_s
```

Comment:
add a comment about why this line is needed (something like "fields of `header` were changed ... ")